numpy/ma/core.py

```diff
@@ -4236,7 +4230,6 @@ def put(self, indices, values, mode='raise'):
         if m is nomask:
             m = getmask(values)
         else:
-            m = m.copy()
```

Comment:
I was questioning the removal of `m.copy`. I assume this is becase `m` is shared. BTW, cannot this now be written
```
if m is nomask:
    blaj
elif getmask(values) is nomask:
    blah
else:
    blah
```